lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1258,6 +1322,22 @@ LogicalResult NodeOp::inferReturnTypes(MLIRContext *context,
   return success();
 }
 
+static LogicalResult verifyRegResetOp(RegResetOp op) {
+  // If the result is a ground type, the async reset value must be a constant
+  // literal or invalid value.
+  if (op.getType().isGround() &&
+      op.resetSignal().getType().isa<AsyncResetType>() &&
```

Comment:
Why is this limited to ground types?  There is a design smell here, have you considered splitting this into two different ops?

Reply:
Even if it's not ground, it should derive from a constant value, yes?  At most it should be subfield/index accesses off a constant?

Reply:
Fully agreed. I was too lazy to type out the full check on the initial push for the PR -- an offense that has been rectified in #1443.